### ISSUE TITLE
DlibOptimizer -- don't hardcode solver_epsilon

### ISSUE THREAD

**Issue by dweindl** [ICB-DCM/pyPESTO#1492]
`DlibOptimizer` currently has a hardcoded `solver_epsilon` (the magic `0.002`):

https://github.com/ICB-DCM/pyPESTO/blob/ead29b36db80d2eeee91b5d29bfbf4430137f8aa/pypesto/optimize/optimizer.py#L672-L678

My current interpretation of that parameter (documentation is a bit sparse) is that it's the minimum absolute objective function improvement (https://github.com/davisking/dlib/blob/6d29e0c7d41dfdffb8c0a0f83995caae84df5083/dlib/global_optimization/global_function_search.cpp#L754). For that, this value is potentially rather high. I am not sure if we should override dlib's default at all (their default is `0`), but at least it should be configurable by the user.

**Comment by PaulJonasJost** [comment]
Agreed, will make it something the user **can** change but with default of 0
